ds1 editor

gucio321 opened the pull request:
fixes #79

Commit:
ds1 editor: rename ds1viewer.go -> ds1_editor_widget.go

Commit:
ds1 editor: rename all DS1ViewerWidget -> DS1Widget

Commit:
update OD2 version; added appropriate data encoder for ds1; added change version feature; ds1 widget uses it's state to build appropriate windows

Commit:
ds1 editor: basic file manager

Commit:
ds1 editor: added object creator

Commit:
ds1 editor: lintfix

Commit:
ds1 editor: paths manager; changed file-delete button into delete image

Commit:
ds1 editor: lintfix

Commit:
ds1 editor: add floor-shadow layout

Commit:
ds1 editor: edit floor, wall and shadow; delete existing floor button (WIP)

gucio321 commented on hswidget/ds1_editor_widget.go:
```diff
@@ -0,0 +1,1004 @@
+		giu.Separator(),
+		giu.Button("Add File##"+p.id+"AddFile").Size(actionButtonW, actionButtonH).OnClick(func() {
+			state.state = ds1EditorStateAddFile
+		}),
```
we can probably call some instance of mpq explorer

Commit:
ds1 editor: lintfix

Commit:
ds1 editor: fixed build errors after merging master

Commit:
ds1 editor: implemented hsutil.MakeImageButton in ds1 editor; createNewFloorShadowRecord and createNewWallRecord methods

Commit:
ds1 editor: broke too long functions

Commit:
ds1 editor: removing floor index in multi-floor ds1's doesn't crash app anymore

Commit:
ds1 editor: minor refactor, broke long methods

gucio321 commented on hswidget/ds1_editor_widget.go:
```diff
@@ -0,0 +1,1115 @@
+	}
+
+	p.ds1.LayerStreamTypes = layerStream
+}
```
We'll be able to remove after merging https://github.com/OpenDiablo2/OpenDiablo2/pull/1059

Commit:
ds1 editor: wall edition methods

Commit:
ds1 editor: moved ds1 widget to its own package

Commit:
ds1 editor: moved ds1 states into separated file

Commit:
ds1 editor: rename ds1ViewerState.state -> mode and moved it into state.go

Commit:
ds1 editor: rename DS1ViewerState -> DS1Widget state

Commit:
ds1 editor:
- moved all stuff responsible for ds1 tiles to tile_records.go
- moved recreateLayerStreamTypes to helper.go

Commit:
ds1 editor: renam ds1_editor_widget.go -> widget.go

Commit:
ds1 editor: rename DS1WidgetState -> DS1State

gravestench commented on hswidget/ds1widget/state.go:
```diff
@@ -0,0 +1,108 @@
+	objX     int32
+	objY     int32
+	objFlags int32
+}
```
is there any reason for these types to be exported?

gucio321 replied:
any one, forgot to change

gravestench commented on hswidget/ds1widget/state.go:
```diff
@@ -0,0 +1,108 @@
+	addObjectState      DS1AddObjectState
+	addPathState        DS1AddPathState
+	addFloorShadowState DS1AddFloorShadowState
+	addWallState        DS1AddWallState
```
yeah, those types probably shouldn't be exported

gucio321 replied:
absolutly

gravestench commented on hswidget/ds1widget/tile_records.go:
```diff
@@ -0,0 +1,207 @@
+				if n != int(state.object) {
+					newWalls = append(newWalls, i)
+				}
+			}
```
probably should change `n` and `i` to something more descriptive

gucio321 replied:
well, n (natural number) is ok, but `i` could be replaced

gravestench replied:
yeah, i'm okay with that. I probably wrote that code to begin with :x

gravestench commented on hswidget/ds1widget/tile_records.go:
```diff
@@ -0,0 +1,207 @@
+				p.ds1.Tiles[y][x].Shadows[0] = newShadow
+			}
+		}
+	}
```
is this setting the `Shadows` for all tiles to the `newShadow` record when you add a new `shadow`... ?

gravestench replied:
what if the tile already had a shadow...?

gucio321 replied:
- when one tile has a shadow, then another tiles too (`DS1.LayerStreamTypes`)
- it is only called, when tile hasn't any shadow

gravestench replied:
Ok, after looking at the tile record struct, i think i get it; tiles have a variable number of walls, floors, and shadows.

gravestench commented on hswidget/ds1widget/helper.go:
```diff
@@ -0,0 +1,49 @@
+// github.com/OpenDiablo2/OpenDiablo2/d2common/d2fileformats/d2ds1.(*DS1).setupLayerStreamType()
+// but this method is unexported for now, so...
+// see https://github.com/OpenDiablo2/OpenDiablo2/pull/1059
+func (p *DS1Widget) recreateLayerStreamTypes() {
```
1059 has been closed; this is a bad idea to put state management code for `d2ds1` outside of `d2ds1`, and an even worse idea to make users of the package responsible for knowing when to call state management/upkeep methods.

we need to take a closer look at `d2ds1` and resolve state management there before we merge this PR.

gravestench reviewed (changes requested):
We need to fix our state management issues in `d2ds1` before we merge this PR. It should not be a responsibility of the user of `d2ds1` to call state management methods when they edit fields in the ds1 struct. State management should be internal to `d2ds1`

Commit:
ds1 editor: minor states are now unexported

Commit:
ds1 editor: repleced some undescriptive single-letter variables in ranges to more descriptive names

Commit:
ds1 editor: editor features for object

Commit:
ds1 editor: refactor: added methods to create trueFalse Combo and input intager

Commit:
ds1 editor: replaced some true-false lists with checkboxes

Commit:
ds1 editor: init of delete object btn

Commit:
ds1 editor: delete object feature

Commit:
ds1 editor: remade Object -> position layout (after replacing numeric labels with inputInt, it looked awful)

Commit:
ds1 editor: add/delete tiles row/col

Commit:
ds1 editor: moved stat management methods into state.go

Commit:
ds1 editor: rename DS1Widget -> widget

Commit:
ds1 editor: rename method DS1Viewer -> Create and returned by it value to interface name (giu.Widget)

Commit:
hsutil: custom widgets: a bit changed panic message (to print invalid
			value type)

Commit:
ds1 editor: added comments for make...Layout, replaced local makeInputInt*/makeCheckBox* with appropriate methods from hsutil

Commit:
ds1 editor: removed add/deleteShadow methods (tiles must hav 1 shadow layer, so we can simply edit it)

Commit:
ds1 editor: lintfix

Commit:
ds1 editor: otile management refactor:

- adding floors/shadows/walls will just add an tile, which we'll edit

- rename DS1EditorState -> widgetState

Commit:
ds1 editor: lintfix

gucio321 committed:
all: buildfix

gucio321 committed:
ds1 editor: lintfix

gucio321 committed:
update od2 version

gucio321 committed:
ds1 editor: fixed build errors caused by new version of OD2

gucio321 committed:
ds1 editor: changing tile grid size by editing width/height fields

gucio321 committed:
ds1 editor: add no-objects-image to hsassets

gucio321 committed:
ds1 editor: adding floor layers

gucio321 committed:
ds1 editor: fixed part of lint errors

gucio321 committed:
ds1 editor: implement a new od2 version:

- create common method for displaying tiles info

gucio321 committed:
ds1 editor: uncomment substitution group code

gucio321 committed:
merge branch 'master' into d2ds1-editor & buildfix

gucio321 committed:
Merge branch 'master' into ds1-editor

gucio321 committed:
ds1 editor:

- cut makeTilesLayout method
- cut unnecesarely long deletePath/Object methods

gucio321 committed:
Merge branch 'ds1-editor' of https://github.com/gucio321/hellspawner into ds1-editor

gucio321 committed:
ds1 editor: update od2 version

gucio321 committed:
ds1 editor: lintfix

gucio321 committed:
all: typos/unnecessary comments/code-cleanup

gucio321 committed:
ds1 editor: lintfix

gucio321 committed:
Merge branch 'master' into ds1-editor

gucio321 committed:
Merge branch 'master' into ds1-editor

gucio321 committed:
Merge branch 'master' into ds1-editor

gucio321 committed:
all: buildfix

gravestench reviewed (approved):
Good work!
